service-api/src/test/java/greencity/dto/habit/AddCustomHabitDtoRequestTest.java

```diff
@@ -0,0 +1,65 @@
+package greencity.dto.habit;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AddCustomHabitDtoRequestTest {
```

Comment:
please add tests for cases when `complexity` is `null` and lower than `min`

Reply:
added